File: Extra/125-Valid-Palindrome.py
# 125. Valid Palindrome

# https://leetcode.com/problems/valid-palindrome/

class Solution:
    def isPalindrome(self, s: str) -> bool:
        import re
        cleaned = re.sub(r'[^a-zA-Z0-9]', '', s)
        a = cleaned.lower()
        return a == a[::-1]

File: 2025-10-08/125-Valid-Palindrome.cpp
//
// Created by egoistpizza on 8.10.2025.
//

// 125. Valid Palindrome
// https://leetcode.com/problems/valid-palindrome/

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {

        int left = 0;
        int right = s.size() - 1;

        while (left < right) {

            while (left < right && !isalnum(s[left])) left++;
            while (left < right && !isalnum(s[right])) right--;

            if (tolower(s[left]) != tolower(s[right])) return false;

            left++;
            right--;
        }
        return true;
    }
};
